### PR TITLE
Bagel: Revenge of the fix

### LINKED ISSUE #15501: Detective's Camera Bugs on Bagel Station doesn't have a mobile viewer, rendering them useless.
(WEB REPORT BY: zyrodil REMOTE: 172.93.109.202:7777)
>Revision (Should be above if you're viewing this from ingame!)

>General description of the issue
There is a cigarette pack in the Detective's office with Camera bugs inside of them. To use them fully, you need a Monitoring device of some sort. But there is no such device to be found in the Detective's office, which means that you have useless Camera Bugs you can put everywhere but you can't use them.

>What you expected to happen
I would expect either no Camera Bugs in the Detective's office or (more preferably) a Video Monitor-device of which would view the video feed of these Camera Bugs.

>What actually happened
I had a package of Camera Bugs for no reason.

>Steps to reproduce if possible
1. Go to the DromedareCo Cigarette pack and pick it up.
2. Ask if one of your security friends wants a cigarette.
3. Pull out Camera Bugs.
4. Search through your office for a Video Device.
5. Fail to find a Video Device.

Thanks for your time, admins. It's not a game breaking thing, it's just wierd that there's Camera Bugs and no way to use them.

### LINKED ISSUE #16195: Bagel SecCheckpoint Map Issue
(WEB REPORT BY: bittervanilla REMOTE: 172.93.109.202:7777)
# Revision
6bdd9c11b81fcf6e805ac9d08f573f8442300f93

# Description
The arrivals security checkpoint has a 'Ready Declaration Device' on the wall with no matching button. Once clicked, it'll endlessly blink, waiting for the deathmatch/thunderdome match to start.
It looks suspicously like Keycard Authentication Device but has none of the functions.
It doesn't even work as a fight button.
HoS office and HoP office appeared to have the correct buttons. Not sure on the others

# What you Expected
No button/a rage cage to appear for your right to board the station/a Keycard Authentication Device

# What Actually Happened
Mild confusion and disappointment

### PR DESCRIPTION
**Pull request by CptWad**
Needs more testing: #15920
fixed #16195
fixed #15501
fixes ais crashing (i think)
:cl: 
 * bugfix: Fixes various things on bagelstation
 * experimental: Possibly fixed AI's crashing as well
 * rscadd: made the cult dungeon harder.

### CODE REVIEW

**Comment by CptWad** [comment]
\>layer
what

File: maps/bagelstation.dmm [hunk 1]
```diff
@@ -11055,7 +11055,7 @@
 "eeE" = (/obj/machinery/light/small{dir = 4; on = 1},/obj/machinery/recharger{pixel_x = 29},/turf/simulated/floor/plating,/area/mine/production)
 "eeF" = (/turf/simulated/wall/r_wall,/area/research_outpost/hallway)
 "eeG" = (/obj/structure/plasticflaps/mining,/obj/machinery/mineral/input,/turf/simulated/floor/plating,/area/research_outpost/hallway)
-"eeH" = (/obj/machinery/conveyor{id_tag = "mining_internal"},/obj/structure/plasticflaps/mining,/obj/machinery/mineral/input,/turf/simulated/floor/plating/airless,/area/research_outpost/hallway)
+"eeH" = (/turf/simulated/floor/airless{tag = "icon-cult"; icon_state = "cult"},/mob/living/simple_animal/hostile/mimic/crate/chest,/turf/simulated/floor/airless{tag = "icon-wood_siding1"; icon_state = "wood_siding1"},/area/mine/unexplored)
```
Comment (Kurfursten) [inline maps/bagelstation.dmm:11058]: Undocumented change

File: maps/bagelstation.dmm [hunk 2]
```diff
@@ -12737,7 +12737,10 @@
 "eKW" = (/obj/item/clothing/under/doomguy,/turf/simulated/floor/airless,/area)
 "eKX" = (/obj/structure/grille/broken,/turf/simulated/floor/plating/airless,/area)
 "eKY" = (/obj/structure/grille,/turf/simulated/floor/airless{broken = 1; icon_state = "floorscorched2"},/area)
-
+"eKZ" = (/obj/effect/mine/kick,/mob/living/simple_animal/hostile/faithless/cult,/turf/simulated/floor/airless{tag = "icon-cult"; icon_state = "cult"},/area/mine/unexplored)
```
Comment (Kurfursten) [inline maps/bagelstation.dmm:12740]: Undocumented change

**Comment by Kurfursten** [comment]
**SNEAKY MAPPERS**

Just document your changes!

**Comment by CptWad** [comment]
its in the super secret cult fight room
I got a complaint it was too easy

**Comment by Kurfursten** [comment]
It's not so hard to just say that upfront, right? I noticed you started doing this when floors turned to glass and a bunch of new vendors appeared after an unrelated mapping PR.

I don't think anyone even objects to having a couple added enemies on the roid but you gotta be honest with people in your OP each time you map. We really try to be transparent when possible, and it's not always easy, but it's worth it.

**Comment by CptWad** [comment]
well sure but its not like I did anything majorly impactful, I just added two mobs and a mine to a room already filled with mobs

**Comment by Shadowmech88** [comment]
"The way in which I broke the rules was innocuous" is not, in my opinion, a good justification for breaking the rules.
Collaborators and coders alike should always strive not only to avoid impropriety, but to avoid the *appearance* of impropriety.

**Comment by CptWad** [comment]
I get that the one guy did the self merge thing  and people are upset about the sneaky coder meme but the main reason it wasnt meantioned was

1. it was told to me on discord and not a git issue
2. I forgot

I understand but I also try to strive to keep the map fluid so that new content can actually see the light of day (like the glass floor tiles and new vendors)
Ill try to be a bit better about it in the future I guess?

**Comment by Shadowmech88** [comment]
It's fine that you forgot. Not ideal, of course, but we're only human.
You just need to add the changes to the OP/changelog, now that you've been reminded.

**Comment by CptWad** [comment]
Was that a demand? Bossy bossy you

**Comment by Shadowmech88** [comment]
I apologize. I sometimes come across as more callous than I intend to via text.
I did not mean to make it sound like a demand.

**Comment by CptWad** [comment]
Sallright its 2am so now I sleep goodnight

**Comment by SonixApache** [comment]
what'd you do to fix the AI crashing?

**Comment by CptWad** [comment]
I think it's because there is both a juke box and a music thingy. Remove the music thingy to see if it solves the problem

**Comment by Aurx** [comment]
>kick fucking mine

M8, you do know what those do?  
They're immulsion breaking as hell.  Don't use them.  If you want to ruin somebody's day without just blowing them up, a stun mine or rad mine will do the trick.

**Comment by SonixApache** [comment]
>kickmines still present

wew

**Comment by PJB3005** [comment]
Whoops.

**Pull request merged**